src/components/app/header/index.js

```diff
@@ -1,11 +1,16 @@
-import React from 'react';
+import React, {useContext} from 'react';
 import './index.css'
+import {CardsContext} from "../CardsContext";
 
 function Header() {
+    const {cards} = useContext(CardsContext);
     return (
         <div className="header header-layout">
             <div className="container">
-                <h1 className="header-title">Header</h1>
+                <div className="header-data">
+                    <h1 className="header-title">Header</h1>
+                    <h6>Cards count <span className="badge badge-secondary">{cards.length}</span></h6>
```

Comment:
Rather than pass all the card array to the header, I would advice to create method in a context to return cards array.